src/common/functions/emailValidation.ts

```diff
@@ -0,0 +1,10 @@
+import { env } from "@/env.mjs";
+
+export const isValidEmail = (email: string) => {
+  const regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  const supportedDomains = env.NEXT_PUBLIC_SUPPORTED_SCH_DOMAINS.split(",");
+  return (
+    regex.test(email) &&
+    supportedDomains.some((domain) => email.endsWith(domain))
+  );
+};
```

Comment:
Thanks for catching this! I have refactored the checking at [b20b011](https://github.com/AfterClass-io/afterclass.io-v2/pull/93/commits/b20b011f1648dc339d7c8b94903242a1ef7dca66)

Also moved the schema to `tools` folder since it is no longer a function